perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2552,16 +2552,71 @@ public void updateFormItemData(PerunSession sess, int appId, ApplicationFormItem
 
 		if (notAllowed.contains(existingData.getFormItem().getType())) throw new RegistrarException("You are not allowed to modify "+existingData.getFormItem().getType()+" type of form items.");
 
+		updateFormItemData(sess, data);
+
+	}
+
+	@Transactional(rollbackFor = Exception.class)
+	public void updateFormItemsData(PerunSession sess, int appId, List<ApplicationFormItemData> data) throws PerunException {
+
+		Application app = getApplicationById(appId);
+
+		if (app == null) throw new InternalErrorException("Application with ID="+appId+" doesn't exist.");
+
+		if (!AuthzResolver.selfAuthorizedForApplication(sess, app)) {
```

Comment:
Just a reminder that by calling only `selfAuthorizedForApplication`, PERUNADMIN will not be allowed to call this method. I suppose it is ok if the whole method will be used only for perun-wui, but it could be insufficient when for example NGUI would want to use it.

Reply:
Thanks for the insight. Use-cases for admin gui are not yet defined, I'll let this to be decided once its needed.